lib/helpers/actions.js

```diff
@@ -0,0 +1,277 @@
+ * @returns {Boolean} Returns whether the action string is valid.
+ */
+function isValidAction(actionString) {
+	return module.exports.actions.some(foundAction => {
+		return foundAction.match.test(actionString);
+	});
+}
+
+/**
+ * Available actions.
+ * @private
+ */
+module.exports.actions = [
+
+	// Action to navigate to a url
+	// E.g. "navigate to http://pa11y.org"
+	{
+		name: 'navigate-url',
+		match: /^navigate to( url)? (.+)$/i,
+		run: async (browser, page, matches) => {
+			const navigateTo = matches[2];
+			try {
+				await page.goto(navigateTo);
+			} catch (error) {
+				throw new Error(`Failed action: Could not navigate to "${navigateTo}"`);
+			}
+		}
+	},
+
+	// Action to click an element
+	// E.g. "click .sign-in-button"
+	{
+		name: 'click-element',
+		match: /^click( element)? (.+)$/i,
+		run: async (browser, page, matches) => {
+			const selector = matches[2];
+			try {
+				await page.click(selector);
+			} catch (error) {
+				throw new Error(`Failed action: no element matching selector "${selector}"`);
+			}
+		}
+	},
+
+	// Action to set an input field value
+	// E.g. "set field #username to example"
+	{
+		name: 'set-field-value',
+		match: /^set( field)? (.+) to (.+)$/i,
+		run: async (browser, page, matches) => {
+			const selector = matches[2];
+			const value = matches[3];
+			try {
+				/* eslint-disable no-shadow */
+				await page.evaluate((selector, value) => {
+					const target = document.querySelector(selector);
+					if (!target) {
+						return Promise.reject(new Error('No element found'));
+					}
+					target.value = value;
+					target.dispatchEvent(new Event('input', {
+						bubbles: true
+					}));
+					return Promise.resolve();
+				}, selector, value);
+				/* eslint-enable no-shadow */
+			} catch (error) {
+				throw new Error(`Failed action: no element matching selector "${selector}"`);
+			}
+		}
+	},
+
+	// Action to check or uncheck a checkbox/radio input
+	// E.g. "check field #example"
+	// E.g. "uncheck field #example"
+	{
+		name: 'check-field',
+		match: /^(check|uncheck)( field)? (.+)$/i,
+		run: async (browser, page, matches) => {
+			const checked = (matches[1] !== 'uncheck');
+			const selector = matches[3];
+			try {
+				/* eslint-disable no-shadow */
+				await page.evaluate((selector, checked) => {
+					const target = document.querySelector(selector);
+					if (!target) {
+						return Promise.reject(new Error('No element found'));
+					}
+					target.checked = checked;
+					target.dispatchEvent(new Event('change', {
+						bubbles: true
+					}));
+					return Promise.resolve();
+				}, selector, checked);
+				/* eslint-enable no-shadow */
+			} catch (error) {
+				throw new Error(`Failed action: no element matching selector "${selector}"`);
+			}
+		}
+	},
+
+	// Action to screen capture the page to a file
+	// E.g. "screen-capture example.png"
+	// E.g. "capture screen to example.png"
+	{
+		name: 'screen-capture',
```

Comment:
I guess this kind of already exists, but not much harm in it being here?

Reply:
i guess not. Tbh I could probably ditch the other one, I didn't really follow through with it anywhere